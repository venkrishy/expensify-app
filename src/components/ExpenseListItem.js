import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (
        <div>
            <p>
                <Link to={`/edit/${id}`}>{description}</Link> - {numeral(amount / 100).format('$0,0.00')} - {moment(createdAt).format("MMMM Do YYYY")} 
            </p>
        </div>
    )
}

export default ExpenseListItem;