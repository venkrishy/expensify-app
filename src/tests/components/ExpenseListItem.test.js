import React from 'react';
import { shallow } from  'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test("should render ExpenseListItem with expenses", () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
    
});

test("should render ExpenseListItem with NO expenses", () => {
    const wrapper = shallow(<ExpenseListItem id={expenses[0].id} />);
    expect(wrapper).toMatchSnapshot();
});

