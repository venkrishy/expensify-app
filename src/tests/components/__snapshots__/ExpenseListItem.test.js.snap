// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with NO expenses 1`] = `
<div>
  <p>
    <Link
      replace={false}
      to="/edit/1"
    />
     - 
    $0.00
     - 
    December 31st 1969
  </p>
</div>
`;

exports[`should render ExpenseListItem with expenses 1`] = `
<div>
  <p>
    <Link
      replace={false}
      to="/edit/1"
    >
      Gum
    </Link>
     - 
    $1.95
     - 
    December 31st 1969
  </p>
</div>
`;
